lib/charms/nginx_ingress_integrator/v0/nginx_route.py

```diff
@@ -0,0 +1,386 @@
+        owasp_modsecurity_custom_rules: configure Nginx ingress
+            integrator owasp-modsecurity-custom-rules option via
+            relation, optional.
+        path_routes: configure Nginx ingress integrator path-routes
+            option via relation, optional.
+        retry_errors: configure Nginx ingress integrator retry-errors
+            option via relation, optional.
+        rewrite_target: configure Nginx ingress integrator
+            rewrite-target option via relation, optional.
+        rewrite_enabled: configure Nginx ingress integrator
+            rewrite-enabled option via relation, optional.
+        service_namespace: configure Nginx ingress integrator
+            service-namespace option via relation, optional.
+        session_cookie_max_age: configure Nginx ingress integrator
+            session-cookie-max-age option via relation, optional.
+        tls_secret_name: configure Nginx ingress integrator
+            tls-secret-name option via relation, optional.
+        nginx_route_relation_name: Specifies the relation name of
+            the relation handled by this requirer class. The relation
+            must have the nginx-route interface.
+    """
+    config: typing.Dict[str, typing.Union[str, int, bool]] = {}
+    if service_hostname is not None:
+        config["service-hostname"] = service_hostname
+    if service_name is not None:
+        config["service-name"] = service_name
+    if service_port is not None:
+        config["service-port"] = service_port
+    if additional_hostnames is not None:
+        config["additional-hostnames"] = additional_hostnames
+    if limit_rps is not None:
+        config["limit-rps"] = limit_rps
+    if limit_whitelist is not None:
+        config["limit-whitelist"] = limit_whitelist
+    if max_body_size is not None:
+        config["max-body-size"] = max_body_size
+    if owasp_modsecurity_crs is not None:
+        config["owasp-modsecurity-crs"] = owasp_modsecurity_crs
+    if owasp_modsecurity_custom_rules is not None:
+        config["owasp-modsecurity-custom-rules"] = owasp_modsecurity_custom_rules
+    if path_routes is not None:
+        config["path-routes"] = path_routes
+    if retry_errors is not None:
+        config["retry-errors"] = retry_errors
+    if rewrite_target is not None:
+        config["rewrite-target"] = rewrite_target
+    if rewrite_enabled is not None:
+        config["rewrite-enabled"] = rewrite_enabled
+    if service_namespace is not None:
+        config["service-namespace"] = service_namespace
+    if session_cookie_max_age is not None:
+        config["session-cookie-max-age"] = session_cookie_max_age
+    if tls_secret_name is not None:
+        config["tls-secret-name"] = tls_secret_name
+
+    _NginxRouteRequirer(
+        charm=charm, config=config, nginx_route_relation_name=nginx_route_relation_name
+    )
+
+
+class _NginxRouteProvider(ops.framework.Object):
```

Comment:
I'm wondering if we shouldn't split requirer and provider in 2 different files like the data platform team is doing ?
See: https://github.com/canonical/data-platform-libs/tree/main/lib/charms/data_platform_libs/v0

Reply:
Maybe, but I think splitting into two different libraries or keeping in the same library are just personal preference. The original `ingress` library keeps everything in the same library, so I am doing it here.